Assets/Scripts/PlayerController.cs

```diff
@@ -1,153 +1,32 @@
-
-    private void ToggleSpecialTrailRenderers(bool toggle)
-    {
-        specialTrailRenderers.ForEach(t =>
-        {
-            t.emitting = toggle;
-        });
-    }
-
-    private void PerformDash()
-    {
-        rb.AddForce(transform.forward * dashForce, ForceMode.Impulse);
-    }
-
-    void FixedUpdate()
-    {
-        if (!EnablePlayerInput)
-        {
-            SetIsWalking(false);
-            return;
-        }
-
-        // early exit, since we cannot do relative-to-camera player movement otherwise
-        if (!CameraManager.Instance.IsActiveCameraValid())
-        {
-            return;
-        }
-
-        Vector3 relativeForward = CleanForwardVector(CameraManager.Instance.GetActiveCameraTransform().forward);
-        Vector3 relativeRight = CalculateRightVector(relativeForward);
-
-        float horizontal = Input.GetAxis("Horizontal");
-        float vertical = Input.GetAxis("Vertical");
-
-        movement = relativeForward * vertical + relativeRight * horizontal;
-        movement.Normalize();
-        bool hasHorizontalInput = !Mathf.Approximately(horizontal, 0f);
-        bool hasVerticalInput = !Mathf.Approximately(vertical, 0f);
-        bool isWalking = hasHorizontalInput || hasVerticalInput;    // to be used later for animations and such
-        SetIsWalking(isWalking);
-
-        HandleControl(movement);
-    }
-
-    public void SetIsWalking(bool isWalking)
-    {
-        IsMoving = isWalking;
-        anim.SetBool(Constants.ANIMATION_BIRDIE_ISWALKING, isWalking);
-    }
-
-    private Vector3 CleanForwardVector(Vector3 forwardVector)
-    {
-        forwardVector.y = 0f;
-        return forwardVector.normalized;
-    }
-
-    private Vector3 CalculateRightVector(Vector3 forwardVector)
-    {
-        // The vector perpendicular to the forward vector and this transform's up, must be the relative right vector
-        return -Vector3.Cross(forwardVector, transform.up).normalized;
-    }
-
-    void HandleControl(Vector3 movementDirection)
-    {
-        Vector3 desiredForward = Vector3.RotateTowards(transform.forward, movementDirection, turnSpeed * Time.fixedDeltaTime, 0f);
-
-        HandleRotation(desiredForward);
-
-        // Only start moving once we are close enough to our desired final rotation (which we are smoothly rotating towards)
-        if ((movementDirection - desiredForward).magnitude < canMoveRotationThreshold)
-        {
-            HandleMovement();
-        }
-    }
-
-    void HandleRotation(Vector3 desiredForward)
-    {
-        rb.MoveRotation(Quaternion.LookRotation(desiredForward));
+        SetControllerConnected();
     }
 
-    void HandleMovement()
+    private void SetControllerConnected()
     {
-        rb.MovePosition(rb.position + movement * (movementSpeed * Time.fixedDeltaTime));
+        string[] controllerNames = Input.GetJoystickNames();
+        
+        if (controllerNames.Length <= 0) return;
+        if (controllerNames.All(string.IsNullOrEmpty)) return;
```

Comment:
In these cases we should set `controllerConnected = false;` right? Especially if we re-call this function?